src/OpenTelemetry.AutoInstrumentation.StartupHook/StartupHook.cs

```diff
@@ -75,20 +77,38 @@ public static void Initialize()
                 string loaderFilePath = Path.Combine(loaderAssemblyLocation, "OpenTelemetry.AutoInstrumentation.Loader.dll");
                 Assembly loaderAssembly = Assembly.LoadFrom(loaderFilePath);
                 loaderAssembly.CreateInstance("OpenTelemetry.AutoInstrumentation.Loader.Startup");
-                StartupHookEventSource.Log.Trace("StartupHook initialized successfully!");
+                Logger.LogInformation("StartupHook initialized successfully!");
             }
             else
             {
-                StartupHookEventSource.Log.Trace("OpenTelemetry.AutoInstrumentation.Instrumentation initialized before startup hook");
+                Logger.LogInformation("OpenTelemetry.AutoInstrumentation.Instrumentation initialized before startup hook");
             }
         }
         catch (Exception ex)
         {
-            StartupHookEventSource.Log.Error($"Error in StartupHook initialization: LoaderFolderLocation: {loaderAssemblyLocation}, Error: {ex}");
+            Logger.LogError($"Error in StartupHook initialization: LoaderFolderLocation: {loaderAssemblyLocation}, Error: {ex}");
             throw;
         }
     }
 
+    private static void InitLogger()
+    {
+        // Workaround to make startup hook load the logger assembly.
+        var executionPath = Assembly.GetExecutingAssembly().Location;
+        var dirname = Path.GetDirectoryName(executionPath);
+        if (!string.IsNullOrEmpty(dirname))
+        {
+            try
+            {
+                Assembly.LoadFrom(Path.Combine(dirname, "OpenTelemetry.AutoInstrumentation.Logging.dll"));
```

Comment:
`OpenTelemetry.AutoInstrumentation.StartupHook.csproj` has project reference to `OpenTelemetry.AutoInstrumentation.Logging.csproj`, hence `OpenTelemetry.AutoInstrumentation.Logging.dll` should get loaded. Is there a reason to load this dynamically?

Reply:
The reason is it does not load it for some reason. My guess is that StartupHook is loaded as `DOTNET_STARTUP_HOOKS` and it just loads single assembly.

Reply:
https://github.com/dotnet/runtime/blob/main/docs/design/features/host-startup-hook.md#no-dependency-resolution-for-non-app-assemblies kudos to @lachmatt for finding it 😉